src/index.ts

```diff
@@ -476,24 +460,74 @@ const plugin: JupyterFrontEndPlugin<void> = {
         }
       }
     });
-    // Track user time, and show a reminder to save the notebook after
-    // five minutes using a toast notification.
-    // Then reset the timer when the notebook is saved manually.
+    // Track user time, and show a reminder to save the notebook once after
+    // five minutes of editing (i.e., once it becomes non-empty and dirty)
+    // using a toast notification.
     let saveReminderTimeout: number | null = null;
+    let isSaveReminderScheduled = false; // a 5-minute timer is scheduled, but it hasn't fired yet
+    let hasShownSaveReminder = false; // we've already shown the toast once for this notebook
+
+    /**
+     * Helper to start the save reminder timer. Clears any existing timer
+     * and sets a new one to show the notification after 5 minutes.
+     */
+    function startSaveReminder(currentTimeout: number | null, onFire?: () => void): number {
+      if (currentTimeout) {
+        window.clearTimeout(currentTimeout);
+      }
+      return window.setTimeout(() => {
+        Notification.info(
+          "It's been 5 minutes since you've been working on this notebook. Make sure to save the link to your notebook to edit your work later.",
```

Comment:
I know this is just copy-pasted from old version, but the wording looks a bit odd.

Also, should the timer be reset when user does save the notebook?

Reply:
Yes, but what if it does not become dirty again. What if a user:
- saves notebook
- goes for a break
- comes back after making no change

will they see a spurious notification?

Reply:
I think it would be okay, as the notification is only to help remind them and is not particularly intrusive either. In this case, how about we opt to change the wording instead of the functionality and act as good Samaritans? :)

Reply:
Or, we could track if they've saved this notebook in the session before already, and in that case, instead of "It's been 5 minutes since you've been working on this notebook", we can say "It's been 5 minutes since you last saved this notebook". What do you think?